libraries/wallet/wallet.cpp

```diff
@@ -1932,6 +1932,32 @@ class wallet_api_impl
       return sign_transaction( tx, broadcast );
    } FC_CAPTURE_AND_RETHROW( (owner_account)(broadcast) ) }
 
+   map<string, son_id_type> list_active_sons()
+   { try {
+           global_property_object gpo = get_global_properties();
+           std::vector<fc::optional<son_object>> son_objects =
+                   _remote_db->get_sons(gpo.active_sons);
+           vector<account_id_type> owners;
+           owners.resize(son_objects.size());
+           std::transform(son_objects.begin(), son_objects.end(), owners.begin(),
+                          [](const fc::optional<son_object>& obj) {
+                                if(!obj)
+                                    FC_THROW("Invalid active SONs list in global properties.");
```

Comment:
What needs to happen in order to get this exception and the one in line 1955? Should we have a test case for this ?

Reply:
I don't know if it is possible really,  you are right, `FC_ASSERT` is better case here.